### PR TITLE
lcms port

### PR DESCRIPTION
**Pull request by Agadoul**
http://www.littlecms.com/

Little CMS intends to be an OPEN SOURCE small-footprint color management engine, with special focus on accuracy and performance. It uses the International Color Consortium standard (ICC), which is the modern standard when regarding to color management. The ICC specification is widely used and is referred to in many International and other de-facto standards. It was approved as an International Standard, ISO 15076-1, in 2005.

### CODE REVIEW

File: ports/lcms/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+)
+
+add_definitions(-DUNICODE -D_UNICODE)
+string(REPLACE "/utf-8" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+
+include_directories(
+"./include"
+)
+
+add_library(lcms ${LIBRARY_TYPE}
```
Comment (traversaro) [inline ports/lcms/CMakeLists.txt]: If you avoid to explicitly indicate the library type in `add_library`, the type is induced by the `BUILD_SHARED_LIBS` CMake variable, that is already set automatically by vcpkg depending on the used tripled. In this way you can get rid of all the logic related to the `SHARED_LIBRARY` option.

Comment (Agadoul) [inline ports/lcms/CMakeLists.txt]: OK. I will update my PR with the change. thanks for the feedback.

**Comment by traversaro** [comment]
I checked a bit the library headers https://github.com/mm2/Little-CMS/blob/master/include/lcms2.h#L221 , and it seems that a bit of logic is necessary to correctly export and import symbols depending on the static/shared configuration. How to plan to handle this? For a possible solution, in `FreeImage` we use a `FreeImageConfig.h`that is different between static and shared configuration, see https://github.com/Microsoft/vcpkg/pull/366/files#diff-d0505d14e4a175bfa2cf0aec61f83131R97 .

**Comment by Agadoul** [comment]
It looks to me that I just need to add_definitions(-DCMS_DLL_BUILD) in the CMakeLists when dynamic is enabled. Isn't that enough?

**Comment by traversaro** [comment]
I may be missing something, but looking in lcms includes in that case the symbols are correctly exported as CMS_DLL_BUILD is defined, but they will not be imported correctly by third party code because CMS_DLL will not be defined.

**Comment by Agadoul** [comment]
Hi, sorry for the long break due to the holiday season. I stayed away from any PC for about 2 weeks. I will resume the PR this week end and have a look at what you mentioned related to the DLL exports.

**Comment by Agadoul** [comment]
Long overdue. Here is the updated PR. I used a patch to fix import/export in lcms2.h when dynamic is being build.

**Comment by ras0219-msft** [comment]
No worries ⛷.

Looks great, thanks for the PR!

**Pull request merged**